lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,50 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    def self.included(base)
+      def base.do_with_sync(*names)
+        names.each do |name|
+          new_name = "#{name}_inner".freeze.to_sym
+          alias_method new_name, name
+          private new_name
+          define_method(name) do |*args, &block|
+            synchronize do
+              method(new_name).call(*args, &block)
+            end
+          end
+        end
+      end
+    end
```

Comment:
`#increment`, gets and sets value
`#reset` delegates to `#value=`
`#value=` should definitely be atomic

For the SlidingWindow ones: 
`"size", "max_size", "resize_to", "<<", "push", "pop", "shift", "unshift", "clear", "first", "last"`, either they are accessors or they modify the array, so they are necessary. 

You **could** make the case to remove the synchronize from the accessors, but why would you make it less safe, for example.
It makes sense to just make it uniform from then on.